packages/web/src/components/composer-input/components/AutocompleteText.tsx

```diff
@@ -0,0 +1,132 @@
+import { useRef } from 'react'
+
+import { SearchCategory, useGetSearchResults } from '@audius/common/api'
+import { SquareSizes, Status, UserMetadata } from '@audius/common/models'
+import { accountSelectors } from '@audius/common/store'
+import {
+  Avatar,
+  Box,
+  Flex,
+  LoadingSpinner,
+  Paper,
+  Popup,
+  Text
+} from '@audius/harmony'
+import { useTheme } from '@emotion/react'
+import { useSelector } from 'react-redux'
+
+import { useProfilePicture } from 'hooks/useUserProfilePicture'
+
+const { getAccountStatus, getUserId } = accountSelectors
+
+type AutocompleteItemProps = {
+  user: UserMetadata
+  onConfirm?: (user: UserMetadata) => void
+}
+
+const AutocompleteItem = ({ user, onConfirm }: AutocompleteItemProps) => {
+  const imageSrc = useProfilePicture(user.user_id, SquareSizes.SIZE_150_BY_150)
+  const { color } = useTheme()
+
+  return (
+    <Flex
+      alignItems='center'
+      p='s'
+      gap='s'
+      borderRadius='s'
+      css={{
+        ':hover': {
+          backgroundColor: color.focus.default,
+          cursor: 'pointer',
+          '*': { color: color.static.white }
+        }
+      }}
+      onClick={() => onConfirm?.(user)}
+    >
+      <Box flex={32}>
+        <Avatar src={imageSrc} w={32} h={32} borderWidth='thin' />
+      </Box>
+      <Flex direction='column' alignItems='flex-start' w='100%'>
+        <Text variant='body' size='s' color='default'>
+          {user.name}
+        </Text>
+        <Text variant='body' size='xs' color='subdued'>
+          {user.handle}
+        </Text>
+      </Flex>
+    </Flex>
+  )
+}
+
+type AutocompleteTextProps = {
+  text: string
+  onConfirm?: (user: UserMetadata) => void
+}
+
+export const AutocompleteText = ({
+  onConfirm,
+  text
+}: AutocompleteTextProps) => {
+  const popupAnchorRef = useRef(null)
+  const searchText = text.slice(1)
+  const accountStatus = useSelector(getAccountStatus)
+  const currentUserId = useSelector(getUserId)
+
+  const params = {
+    query: searchText,
+    category: 'users' as SearchCategory,
+    currentUserId,
+    limit: 6,
+    offset: 0
+  }
+
+  const { data, status } = useGetSearchResults(params, {
+    debounce: 500,
```

Comment:
500 seems like a lot tbh but also maybe thats just right? Looking forward to playing around with it and seeing how it feels!

Reply:
This was from Ray's testing, but yea it doesn't feel that long when typing unless you're trying really quick

Reply:
my 500ms was to make it feel like something was happening when things were already cached in the link unfurl. not sure if we need any debounce here. iiuc, this just searches as you're typing? Or was this value lifted from search autocomplete?